go.mod

```diff
@@ -1,28 +1,33 @@
 module github.com/containerd/nerdctl/v2
 
-go 1.21.12
+go 1.22.0
+
+// FIXME:
+// github.com/docker/docker/pkg/sysinfo has been replaced by a fork kept under ./pkg2/sysinfo
+// as Moby is not going to move to containerd v2 anytime soon or fix these transient dependencies.
+// We should still move back to upstream in the future, and remove our copy.
 
 require (
 	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/Microsoft/go-winio v0.6.2
 	github.com/Microsoft/hcsshim v0.12.5
 	github.com/compose-spec/compose-go/v2 v2.1.6
-	github.com/containerd/accelerated-container-image v1.1.4
+	github.com/containerd/accelerated-container-image v1.1.5-0.20240712041225-3c4dd559fed4
 	github.com/containerd/cgroups/v3 v3.0.3
 	github.com/containerd/console v1.0.4
-	github.com/containerd/containerd v1.7.20
-	github.com/containerd/containerd/api v1.7.19
+	github.com/containerd/containerd/api v1.8.0-rc.2
+	github.com/containerd/containerd/v2 v2.0.0-rc.3
 	github.com/containerd/continuity v0.4.3
 	github.com/containerd/errdefs v0.1.0
 	github.com/containerd/fifo v1.1.0
 	github.com/containerd/go-cni v1.1.10
-	github.com/containerd/imgcrypt v1.1.11
+	github.com/containerd/imgcrypt v1.2.0-rc1.0.20240709223013-f3769dc3e47f
 	github.com/containerd/log v0.1.0
-	github.com/containerd/nydus-snapshotter v0.14.0
+	github.com/containerd/nydus-snapshotter v0.14.1-0.20240806063146-8fa319bfe9c5
 	github.com/containerd/platforms v0.2.1
-	github.com/containerd/stargz-snapshotter v0.15.1
-	github.com/containerd/stargz-snapshotter/estargz v0.15.1
-	github.com/containerd/stargz-snapshotter/ipfs v0.15.1
+	github.com/containerd/stargz-snapshotter v0.15.2-0.20240709063920-1dac5ef89319
+	github.com/containerd/stargz-snapshotter/estargz v0.15.2-0.20240709063920-1dac5ef89319
+	github.com/containerd/stargz-snapshotter/ipfs v0.15.2-0.20240709063920-1dac5ef89319
```

Comment:
why do we need to change these deps to a tmp versions? Is it because they haven't had an official release that uses containerd v2?

Reply:
Correct. I did submit PRs to these dependencies to incorporate containerd v2 RC3, but there is no release (yet).